### PR TITLE
Add the bbtools bbduk tool

### PR DESCRIPTION
**Pull request by gregvonkuster**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/bbtools/bbduk.xml [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+                            <option value="l">Trim to the left</option>
+                        </param>
+                        <param name="minlength" type="integer" value="10" label="Minimum read length" help="Trimmed reads shorter than this will be discarded, pairs will be discarded if both are shorter."/>
+                     </when>
+                </conditional>
+            </when>
+            <when value="no"/>
+        </conditional>
+        <conditional name="reset_defaults_cond">
+            <param name="reset_defaults"  type="select" label="Reset default parameter values?">
```
[the remark this answers is not in the document]
Comment (gregvonkuster) [reply tools/bbtools/bbduk.xml]: I've changed the label to the more heavily used "Advanced options".

File: tools/bbtools/bbduk.xml [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+                <option value="no" selected="true">No</option>
+                <option value="yes">Yes</option>
+            </param>
+            <when value="no"/>
+            <when value="yes">
+                <param name="k" type="integer" value="27" min="1" label="Kmer length used for finding contaminants"/>
```
[the remark this answers is not in the document]
Comment (gregvonkuster) [reply tools/bbtools/bbduk.xml]: "name" has been changed to "argument" in all the places it should be I think.

**Comment by bgruening** [comment]
> Applying linter inputs... FAIL
.. ERROR: Select [reference] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
.. ERROR: Select [outputs_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
.. ERROR: Select [output_stats_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
.. ERROR: Select [output_hists_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute

Some more linting hints which we got with the new Galaxy :)

**Comment by gregvonkuster** [comment]
> > Applying linter inputs... FAIL
> > .. ERROR: Select [reference] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
> > .. ERROR: Select [outputs_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
> > .. ERROR: Select [output_stats_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
> > .. ERROR: Select [output_hists_select] `display="checkboxes"` is incompatible with `optional="false"`, remove the `display` attribute
> 
> Some more linting hints which we got with the new Galaxy :)

Yeah, the other 2 bbtools that used to pass linting no longer pass.  How do I get my planemo environment to catch these new linting errors?  I've updated to the latest version, but it still doesn't catch these.

**Comment by gregvonkuster** [comment]
I've now got my local planemo environment updated and working, so it now detects these new linting errors.

**Comment by gregvonkuster** [comment]
I believe this is ready for final review / merge.  Thanks!

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @gregvonkuster

**Pull request merged**